test/EndToEndTests/Common/Microsoft.OData.Client.E2E.TestCommon/PriorityOrderer.cs

```diff
@@ -0,0 +1,52 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="PriorityOrderer.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved.
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Microsoft.OData.Client.E2E.TestCommon
+{
+    /// <summary>
+    /// Orders test cases based on their <see cref="TestPriorityAttribute"/> priority value.
+    /// </summary>
+    public class PriorityOrderer : ITestCaseOrderer
+    {
+        public IEnumerable<TTestCase> OrderTestCases<TTestCase>(IEnumerable<TTestCase> testCases) where TTestCase : ITestCase
+        {
+            var sortedMethods = new SortedDictionary<int, List<TTestCase>>();
+
+            foreach (TTestCase testCase in testCases)
+            {
+                int priority = 0;
+
+                foreach (IAttributeInfo attr in testCase.TestMethod.Method.GetCustomAttributes((typeof(TestPriorityAttribute).AssemblyQualifiedName)))
+                    priority = attr.GetNamedArgument<int>("Priority");
+
+                GetOrCreate(sortedMethods, priority).Add(testCase);
+            }
+
+            foreach (var list in sortedMethods.Keys.Select(priority => sortedMethods[priority]))
+            {
+                list.Sort((x, y) => StringComparer.OrdinalIgnoreCase.Compare(x.TestMethod.Method.Name, y.TestMethod.Method.Name));
+                foreach (TTestCase testCase in list)
+                    yield return testCase;
+            }
+        }
+
+        static TValue GetOrCreate<TKey, TValue>(IDictionary<TKey, TValue> dictionary, TKey key) where TValue : new()

```

Comment:
nitpick: I wonder why this had to be a generic method yet it's only used once, and only sed against known key of and value types `int` and `List<TTestCase>>`. I think that made slightly harder to read. But not a big deal.